### PR TITLE
chore: Optimize CircleCI workflow

### PR DESCRIPTION
**Pull request by sawadashota**
## Proposed changes

Optimize CircleCI workflow to catch error and end faster.

CircleCI jobs work 4 jobs maximum at same time because of free OSS plan.
In hydra, there are 12 jobs to be kicked at created PR but actually just each random 4 jobs proceed.

I think there are 2 problems.
First, we cannot catch error faster if similar and long jobs such as `test-e2e-*` or `test-legacy-*` are kicked.
Second, error PR blocks other PR because concurrency limitation is shared. If on of `test-e2e-*` failed, others are likely to fail for same reason.

So I changed workflow like this

before
image

after
image

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

**Comment by aeneasr** [comment]
Thank you! The problem really is CircleCI here which doesn't support canceling parallel jobs. Do you know if GitHub Actions does that? I think that the wait time is caused by the "executor" machine type which we currently use to check the legacy migrations. I was thinking of removing these tests once 1.5.2 is out because, unless we make changes, I don't think that there will be any issues with the new migration system.

**Comment by aeneasr** [comment]
I just checked and it appears that GitHub actions supports up to 20 concurrent workflows per repository, which would be much much more than we have on CircleCI right now. Since it's free for open source without hard usage limits (as opposed to CircleCI) it might be something we need to switch to. Would be a lot of work though :/

**Comment by sawadashota** [comment]
Yeah, GitHub Actions seems better and it will need  a lot of effort.
I couldn't find canceling parallel jobs spec or action but it seems it's possible if we use GitHub API.
https://developer.github.com/v3/actions/

File: .circleci/config.yml [hunk 1]
```diff
@@ -217,12 +227,10 @@ workflows:
             - nancy/test
             - test
             - golangci/lint
-            - test-e2e-memory
             - test-e2e-postgres
             - test-e2e-mysql
             - test-e2e-cockroach
             - test-e2e-plugin
-            - test-legacy-migrations-postgres
```
Comment (aeneasr) [inline .circleci/config.yml:225]: Why did you remove these?

**Comment by aeneasr** [comment]
Yeah I also saw that it's not possible to do manual job approval, which we currently use for sending out the newsletter :/

File: .circleci/config.yml (hunk 1 above)
Comment (sawadashota) [reply .circleci/config.yml:225]: Because it still depend on indirectly. `test-legacy-migrations-mysql` and `test-legacy-migrations-cockroach` depend on `test-legacy-migrations-postgres`. So does `test-e2e-memory`.

**Comment by aeneasr** [comment]
Thank you!
